bench/gemm.h

```diff
@@ -9,37 +9,111 @@
 #pragma once
 
 #include <benchmark/benchmark.h>
+#include <cstdint>
+#include <initializer_list>
+#include <functional>
+#include <string>
+#include <vector>
+
+class BenchmarkWrapper {
+  public:
+    BenchmarkWrapper(benchmark::internal::Benchmark* benchmark, bool blockwise)
+      : benchmark_(benchmark), blockwise_(blockwise) {}
+
+    void ArgNames(const std::initializer_list<std::string>& names) {
+      std::vector<std::string> vals(names.begin(), names.end());
+      if (blockwise_ && names.size() < 4) {
+        vals.push_back("BL");
+      }
+      benchmark_->ArgNames(vals);
+    }
+
+    void Args(const std::initializer_list<int64_t>& args) {
+      std::vector<int64_t> vals(args.begin(), args.end());
+      if (blockwise_ && args.size() < 4) {
+        // Blockwise kernel testers interpret SIZE_MAX as rounded K.
+        vals.push_back(SIZE_MAX);
+      }
+      benchmark_->Args(vals);
+    }
+
+  private:
+    benchmark::internal::Benchmark* benchmark_;
+    bool blockwise_;
+};
+
+void Configure(benchmark::internal::Benchmark* benchmark, void (setup_method)(BenchmarkWrapper*), bool blockwise) {
```

Comment:
Not enough power in macros, eh? LOL

Reply:
Yeah, I played around with it a bit, but this was the most minimal solution I could find. It ends up being a bit trickier than I expected due to the technical constraints, but this way requires almost no changes to the existing arg defs, which I like.